test/unit/sessions/test_attachment_models.py

```diff
@@ -0,0 +1,513 @@
+        assert "outputRelativeDirectories" not in manifest_data
+
+        assert result_dict["localManifestPaths"] == []
+        assert result_dict["localRootPath"] == "/local/root"
+
+    def test_from_dict_deserialization(self):
+        """Test creating WorkerManifestProperties from dictionary."""
+        # GIVEN
+        data = {
+            "manifestProperties": {
+                "rootPath": "/source/path",
+                "rootPathFormat": "posix",
+                "fileSystemLocationName": "shared_storage",
+                "inputManifestPath": "input.json",
+                "inputManifestHash": "hash123",
+                "outputRelativeDirectories": ["out1", "out2"],
+            },
+            "localManifestPaths": ["/local/manifest1.json", "/local/manifest2.json"],
+            "localRootPath": "/local/root",
+        }
+
+        # WHEN
+        worker_props = WorkerManifestProperties.from_dict(data)
+
+        # THEN
+        # Verify manifest properties
+        assert worker_props.root_path == "/source/path"
+        assert worker_props.root_path_format == PathFormat.POSIX
+        assert worker_props.file_system_location_name == "shared_storage"
+        assert worker_props.input_manifest_path == "input.json"
+        assert worker_props.input_manifest_hash == "hash123"
+        assert worker_props.output_relative_directories == ["out1", "out2"]
+
+        # Verify other fields
+        assert worker_props.local_manifest_paths == [
+            "/local/manifest1.json",
+            "/local/manifest2.json",
+        ]
+        assert worker_props.local_root_path == "/local/root"
+
+    def test_from_dict_with_missing_optional_fields(self):
+        """Test from_dict with missing optional fields."""
+        data = {
+            "manifestProperties": {
+                "rootPath": "/source/path",
+                "rootPathFormat": "windows",
+            },
+            "localRootPath": "/local/root",
+        }
+
+        worker_props = WorkerManifestProperties.from_dict(data)
+
+        assert worker_props.root_path == "/source/path"
+        assert worker_props.root_path_format == PathFormat.WINDOWS
+        assert worker_props.file_system_location_name is None
+        assert worker_props.input_manifest_path is None
+        assert worker_props.input_manifest_hash is None
+        assert worker_props.output_relative_directories is None
+        assert worker_props.local_manifest_paths == []  # Default empty list
+        assert worker_props.local_root_path == "/local/root"
+
+    def test_from_dict_missing_required_fields_raises_error(self):
+        """Test that from_dict raises KeyError for missing required fields."""
+        # GIVEN/WHEN/THEN - Missing manifestProperties
+        data_missing_manifest = {
+            "localRootPath": "/local/root",
+        }
+        with pytest.raises(KeyError):
+            WorkerManifestProperties.from_dict(data_missing_manifest)
```

Comment:
a little unusual in the way pytest does this, but how about this:


```suggestion
        # GIVEN - Missing manifestProperties
        data_missing_manifest = {
            "localRootPath": "/local/root",
        }
        # THEN
        with pytest.raises(KeyError):
            # WHEN
            WorkerManifestProperties.from_dict(data_missing_manifest)
```

same applies to the tests below